Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/SlimeAnimation.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SlimeAnimation : ISlimeComponent
+{
+    
+    [SerializeField] private GameObject _explosionParticles;
+    
+    [SerializeField] private float _squashAmount = 0.8f;
+    [SerializeField] private float _stretchAmount = 1.2f;
+    [SerializeField] private float _animationSpeed = 5f;
+
+    private bool _doSquashing = true;
+    private bool _isSquashing = false;
+
+
+    private void Update()
+    {
+        if (_doSquashing)  
+        {
+            if (!_isSquashing)
+            {
+                StartCoroutine(SquashAndStretch());
+            }
+        }
+    }
+
+    public void SpawnExplosionParticles()
+    {
+        Instantiate(_explosionParticles, transform.position, Quaternion.identity);
+    }
+
+    public void ActivateAnimation()
+    {
+        _doSquashing = true;
+    }
+
+    public void DeactivateAnimation()
+    {
+        _doSquashing = false;
+    }
+    
+    
+    IEnumerator SquashAndStretch()
+    {
+        _isSquashing = true;
+
+        // Squash
+        while (transform.localScale.y > _squashAmount)
+        {
+            transform.localScale -= new Vector3(0f, Time.deltaTime * _animationSpeed, 0f);
```

Comment:
Also, I don't like that you directly references the transform here. This will always take the transform to which this component is attached to, meaning that if we added to the wrong object it won't work properly.
Instead I would use pass a Transform reference in the Configure() method. The mediator should be the one who gives you the transform.